bin/cheritest/cheritest_cheriabi.c

```diff
@@ -237,3 +237,17 @@ test_cheriabi_mmap_unrepresentable(const struct cheri_test *ctp __unused)
 
 	cheritest_success();
 }
+
+void
+test_cheriabi_malloc_zero_size(const struct cheri_test *ctp __unused)
+{
+	void *cap;
+
+	cap = malloc(0);
+	if (cap == NULL)
+		cheritest_failure_errx("malloc(0) returned NULL");
```

Comment:
I don't like this test, it'll break if you ever swap the allocator for a perfectly safe and standards-compliant one that returns NULL for malloc(0)... we should permit NULL in my opinion.